### ISSUE TITLE
Implement portable installation option

### ISSUE THREAD

**Issue by ExE-Boss** [Nexus-Mods/Vortex#106]
Instead of storing the data in <code><var title="Using the PowerShell environment variable syntax">$env:appdata</var>/Vortex</code>, a portable installation would store all the data in the installation directory (ex. a flash drive).

**Comment by Corgano** [comment]
Please add a portable flag / option. I want to have multiple versions of a game installed with different sets of mods, and vortex does not allow this. having a portable installation would let me have multiple versions of vortex i could easily use one per mod set i desire.

**Comment by festro** [comment]
> Please add a portable flag / option. I want to have multiple versions of a game installed with different sets of mods, and vortex does not allow this. having a portable installation would let me have multiple versions of vortex i could easily use one per mod set i desire.

I think vortex uses game profiles for different configurations however it's not enabled by default and I haven't used it enough to know what can be done with it.

**Comment by Chaython** [comment]
Been months, any update?
Really want a portable edition. w/ portabledat rather than appdata.

**Comment by festro** [comment]
Yeah this is continuing to be the greatest thorn in my side with vortex. Once again I'm left at that ultimatum of reinstalling windows for that boost in performance, or save myself the headache that is playing musical chairs and jumping hoops with vortex's directories to get my mods working this is ridiculous I never remembered having this issue with NMM.

**Comment by Chaython** [comment]
Vortex-master\src\app\Application.ts (3 hits)
Line 427:       : path.join(process.env['APPDATA'], vortexPath, currentStatePath);
Line 462:       : path.join(process.env['APPDATA'], vortexPath, currentStatePath);
Line 497:       : path.join(process.env['APPDATA'], vortexPath, currentStatePath);
Vortex-master\Vortex-master\src\extensions\nexus_integration\texts.ts (1 hit)
Line 9:       .catch(() => '%LOCALAPPDATA%\\Google\\Chrome\\User Data')
Vortex-master\src\extensions\nexus_integration\util\chromePath.ts (3 hits)
Line 16:   const appPath = remote.app.getPath('appData');
Line 18:     const userData = process.env.LOCALAPPDATA !== undefined
Line 19:       ? path.join(process.env.LOCALAPPDATA, 'Google', 'Chrome', 'User Data')
Vortex-master\src\util\EpicGamesLauncher.ts (2 hits)
Line 53:           'AppDataPath');
Line 166:       const configPath = path.join(process.env['LOCALAPPDATA'], 'EpicGamesLauncher', 'Saved', 'Config', 'Windows', 'GameUserSettings.ini');
Vortex-master\src\util\Steam.ts (2 hits)
Line 234:                 .map(appData => {
Line 235:                   const { name, manifestData } = appData;

Probably just need to change Application.ts to represent a relative path within the vortex folder, then build it yourself; if it was such a big deal. Though for save games and mod dirs it's much more work than that.
I use a 20gb volume for my windows directory. But I just make a bunch of symbolic links for most apps. You could make a batch file and map the drive as a dir rather than a letter so windows won't randomly change the location.
________________________________

**Comment by darklinkpower** [comment]
This would be very helpful, I have an external hard drive and it would help having a portable version that stores all data in it without having to manually configure paths.

**Comment by gmcdevitt** [comment]
I'd also really like a portable option for moving between computers and being able to fully backup vortex.

**Comment by Greg-Nexus** [comment]
Added to Product Backlog: https://gitlab.nexdev.uk/vortex/vortex/-/issues/150

**Comment by modderchris** [comment]
I saw the --user-data as a command line option but what is the syntax? is it even real?
the only other thing by greg-nexus seems to point to another type of software by a similar name but not "vortex mod manager"
I really need my vortex install to sync both mod installs and mod staging to the USB drive as I use a laptop and want to also run on my desktop sometimes
I have been racking my brains on how to have the staging sync 100% on a USB drive as well as sync the deployment to two separate pc's(storing the game on the usb would be fine too) but "vortex mod manager"  seems to have no documenation of command line pre-launch tricks even on the bloody site they shill it on
I have tried using a shortcut with a --portable as that should exist as a simple way to store local cache in the installdir/userdata subdir just like dolphin and then the portable can link to a wildcard like staging dir and such using directory symlinks
on each pc you just symlink the folder the portable links to to whatever drive the game is actually on then have a folder called "modstage" and one with the game on that drive
I am still baffled by the lack of seperate drive deployments but whatever as I use symlinks in windows to link other game configs to the install dir rather than %userprofile% folders as that is just a dumb trend that games need to stop doing
set up named save folders in a non-%userprofile% directory but don't lose my saves/configs when windows inevitably dies and needs a reinstall
this is literally a CRITICAL feature that either doesn't exist or REQUIRES documentation that is missing
FIX IT NOW
